### PR TITLE
Include both cross-file/interfile and cross-function/interprocedural

### PR DESCRIPTION
**Pull request by adamkvitek**
Note for @s-santillan and @pabloest (this has been originally voiced by @IagoAbal):

Can we make the two groups of words double-way synonyms (for example - a user will search for "crossfile" and get our interfile and cross-file terms results)? This change should affect marketing and docs websites. The two groups are:

- cross-file/crossfile/interfile/inter-file (one synonym group)
- cross-function/crossfunction/across-function/interprocedural/inter-procedural (another synonym group)

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ Both Semgrep and SonarQube use static analysis to find bugs, but there are a few
...
 - Semgrep is LGPL-2.1, SonarQube offers an open-source version but it is missing features. For example, 12 of the supported languages are not available in the open-source offering, and more powerful dataflow features are only available in the paid versions.
 - Semgrep supports user-defined autofixes; SonarQube does not.
-- Semgrep focuses on speed and ease-of-use, making analysis possible at up to 20K-100K loc/sec per rule. SonarQube authors [report approximately 0.4K loc/sec for rulesets in production](https://web.archive.org/web/20210127020636/https://community.sonarsource.com/t/performance-guide-for-large-project-analysis/148/2).
+- Semgrep focuses on speed and ease-of-use, making analysis possible at up to 20K-100K loc/sec per rule. SonarQube authors [report approximately 0.4K loc/sec for rulesets in production](https://web.archive.org/web/20221109203440/https://community.sonarsource.com/t/performance-guide-for-large-project-analysis/148/2).
```
Comment (adamkvitek) [inline docs/faq.md:136]: I am not sure that this link should stay...

Comment (milanwilliams) [reply docs/faq.md:133]: I think if we're to mention this - we should keep a citation / link. If you want to remove the link, we should remove this whole line, IMO!

Comment (adamkvitek) [reply docs/faq.md:133]: The old link is broken now https://web.archive.org/web/20210127020636/https://community.sonarsource.com/t/performance-guide-for-large-project-analysis/148/2 After searching, the https://web.archive.org/web/20221109203440/https://community.sonarsource.com/t/performance-guide-for-large-project-analysis/148/2 seems like the correct replacement. So it should be fine...

**Review by IagoAbal — APPROVED** [review]

File: docs/release-notes/may-2023.md [hunk 2]
```diff
@@ -130,9 +130,9 @@ These rules are available through the [Semgrep Registry](https://semgrep.dev/exp
     - [Dataflow traces in PR comments](/semgrep-cloud-platform/github-pr-comments/#dataflow-traces-in-pr-comments)
     - [Dataflow traces in MR comments](/semgrep-cloud-platform/gitlab-mr-comments/#dataflow-traces-in-mr-comments)
 - Taint analysis:
-    - Taint labels now mostly work interprocedurally (cross-function analysis), except for labeled propagators.
+    - Taint labels now mostly work cross-function (interprocedural)ly (cross-function analysis), except for labeled propagators.
```
Comment (s-santillan) [inline docs/release-notes/may-2023.md:133]:
```suggestion
    - Taint labels now mostly work cross-function (interprocedural)ly (cross-function analysis), except for labeled propagators.
```

Looks like (interprocedural)ly is a bug?

**Review by s-santillan — APPROVED** [review]
LGTM, I'll make a separate PR for the synonyms

**Pull request merged**